Leanplum/Api/LPRegisterDeviceApi.m

```diff
@@ -0,0 +1,57 @@
+//
+//  LPRegisterDeviceApi.m
+//  Leanplum
+//
+//  Created by Grace on 5/14/19.
+//  Copyright © 2019 Leanplum. All rights reserved.
+//
+
+#import "LPRegisterDeviceApi.h"
+#import "LPWSManager.h"
+#import "LPConstants.h"
+#import "LPApiConstants.h"
+#import "LPAPIConfig.h"
+#import "LPErrorHelper.h"
+
+@implementation LPRegisterDeviceApi
+
++ (void) registerDevice:(NSDictionary *)attributes
+                success:(void (^)(void))success
+                failure:(void (^)(NSError *error))failure {
+    
+    void (^successResponse) (NSDictionary *) = ^(NSDictionary *response) {
+        NSError *error = nil;
+        NSArray *responseArray = [response valueForKey:@"response"];
+        NSDictionary *resultDict = responseArray[0];
+        if (error != nil) {
+            failure(error);
+        }
+        else {
+            if ([resultDict objectForKey:@"success"]) {
+                success();
+            } else {
+                NSError *error = [LPErrorHelper makeResponseError:@{@"message": @"Invalid Input"}];
```

Comment:
We need to revisit this method . makeResponseError. 
The real usecase has to be that we pass the http error to this function and then it creates a response for us. Not, that we set error and message here.

Reply:
changed makeUserInfo, now default error message should be attached if no error message is received from server